src/tests/src/libs/antares/study/short-term-storage-input/test-on-making-hour-groups-from-string.cpp

```diff
@@ -0,0 +1,169 @@
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/unit_test.hpp>
+
+#include "antares/study/parts/short-term-storage/makeGroupsOfHoursFromString.h"
+
+using namespace Antares::Data::ShortTermStorage;
+
+BOOST_AUTO_TEST_SUITE(test_of_making_groups_of_hours_from_a_string)
+
+BOOST_AUTO_TEST_CASE(first_quite_simple_hours_field)
+{
+    std::string hoursField = "[1, 2], [3, 4]";
+
+    const auto groupsOfHours = makeGroupsOfHours(hoursField);
+
+    const std::vector<std::set<int>> expected_result = {{1, 2}, {3, 4}};
+    BOOST_CHECK(groupsOfHours == expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(hours_field_has_spaces_inside_brackets__after_a_comma_separator)
+{
+    std::string hoursField = "[5,   1]";
+
+    const auto groupsOfHours = makeGroupsOfHours(hoursField);
+
+    const std::vector<std::set<int>> expected_result = {{5, 1}};
+    BOOST_CHECK(groupsOfHours == expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(hours_field_has_spaces_right_before_opening_bracket)
+{
+    std::string hoursField = "[  7, 6]";
+
+    const auto groupsOfHours = makeGroupsOfHours(hoursField);
+
+    const std::vector<std::set<int>> expected_result = {{7, 6}};
+    BOOST_CHECK(groupsOfHours == expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(hours_field_has_spaces_right_before_closing_bracket)
+{
+    std::string hoursField = "[9, 0  ]";
+
+    const auto groupsOfHours = makeGroupsOfHours(hoursField);
+
+    const std::vector<std::set<int>> expected_result = {{9, 0}};
+    BOOST_CHECK(groupsOfHours == expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(hours_field_has_spaces_between_groups)
+{
+    std::string hoursField = "[5,1] ,  [8,7]";
+
+    const auto groupsOfHours = makeGroupsOfHours(hoursField);
+
+    const std::vector<std::set<int>> expected_result = {{5, 1}, {8, 7}};
+    BOOST_CHECK(groupsOfHours == expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(hours_field_has_spaces_before_a_group)
+{
+    std::string hoursField = "  [5,1], [8,7]";
+
+    const auto groupsOfHours = makeGroupsOfHours(hoursField);
+
+    const std::vector<std::set<int>> expected_result = {{5, 1}, {8, 7}};
+    BOOST_CHECK(groupsOfHours == expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(hours_field_has_spaces_everywhere)
+{
+    std::string hoursField = " [  5,   1] , [ 8, 7   ]  ";
+
+    const auto groupsOfHours = makeGroupsOfHours(hoursField);
+
+    const std::vector<std::set<int>> expected_result = {{5, 1}, {8, 7}};
+    BOOST_CHECK(groupsOfHours == expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(hours_field_contains_groups_of_different_sizes)
+{
+    std::string hoursField = "[5, 1, 98], [8, 79, 54, 32], [109]";
+
+    const auto groupsOfHours = makeGroupsOfHours(hoursField);
+
+    const std::vector<std::set<int>> expected_result = {{5, 1, 98}, {8, 79, 54, 32}, {109}};
+    BOOST_CHECK(groupsOfHours == expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(hours_field_contains_unwnated_but_ignored_chars_between_brackets)
+{
+    std::string hoursField = "[5, 1, 23], {I will be ignored}[4, 19]";
+
```

Comment:
this is less restrictive than the original code

Reply:
The original unit tests were modified and integrated.
So there are simply more tests now, isn't it ?